### PR TITLE
AbsSurvey -- Beginnings

### PR DESCRIPTION
**Pull request by profxj**
AbsSurvey class

Pretty bare bones but includes:
test
docs 
notebook

### CODE REVIEW

File: linetools/isgm/abssurvey.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+        abs_type : str
+          Type of AbsSystem in the Survey, e.g.  MgII, DLA, LLS
+        nsys : int, optional
+          Number of AbsSystem in the Survey
+        ref : string, optional
+          Reference(s) for the survey
+        """
+        self.abs_type = abs_type
+        self.ref = ref
+        self._abs_sys = []
```
Comment (nhmc) [inline linetools/isgm/abssurvey.py]: Why does this have a leading underscore?

File: linetools/isgm/abssurvey.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+                    t.add_row( row )
+                continue
+            else:
+                raise ValueError('Multiple entries...')
+
+        # Return
+        return t[1:]
+
+    # Mask
+    def upd_mask(self, msk, increment=False):
```
Comment (nhmc) [inline linetools/isgm/abssurvey.py]: I prefer `update_mask`

File: linetools/isgm/abssurvey.py [hunk 3]
```diff
@@ -0,0 +1,325 @@
+""" Class for and AbslineSurvey
+"""
+
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+import numpy as np
+import imp, json, copy
+from abc import ABCMeta
```
Comment (nhmc) [inline linetools/isgm/abssurvey.py]: This import can be removed (also please check for any other unused imports)

Comment (nhmc) [inline linetools/isgm/abssurvey.py]: Oops, I now see it is used AbslinSurvey. Disregard :)

File: linetools/isgm/abssurvey.py [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    """Translate abstype into Class
+    Parameters:
+    ----------
+    abstype: str
+      AbsSystem type, e.g. 'LLS', 'DLA'
+
+    Returns:
+    --------
+    Class name
+    """
```
Comment (nhmc) [inline linetools/isgm/abssurvey.py]: Please make docstrings sphinx-friendly.

**Comment by nhmc** [comment]
I left some inline comments.  Also, it seems like nsys has to be specified in addition to adding each system? If so it would be much better to make nsys a `@property` that returns `len(_abs_sys)` when called.

File: linetools/isgm/abssurvey.py (hunk 1 above)
Comment (profxj) [inline linetools/isgm/abssurvey.py]: It is very rare that one will access this list of objects directly.

I could, however, name it _abssys if that is easier to read.

File: linetools/isgm/abssurvey.py (hunk 2 above)
Comment (profxj) [inline linetools/isgm/abssurvey.py]: Done

File: linetools/isgm/abssurvey.py (hunk 1 above)
Comment (nhmc) [inline linetools/isgm/abssurvey.py]: But in the example notebook you access it directly to add new systems. If that's not the intended usage, the example should be changed.  (Also `_abs_sys`  is way better than `_abssys`)

**Comment by profxj** [comment]
I like the property idea and have put it in.

I may have had it in mind to allows a Survey without filling
the abs_sys list but nevermind that.

**Comment by nhmc** [comment]
I see you fixed the docs too. Please merge when ready.

**Comment by profxj** [comment]
Merging

**Pull request merged**